backend/ReservationApp/views.py

```diff
@@ -50,3 +52,24 @@ def post(self, request):
         send_confirmation_email(start_time, user.email, restaurant)
 
         return make_success_response()
+
+class ReservationUpdateStateView(APIView):
+    def put(self, request):
+        try:
+            restaurant_id = request.data['restaurant_id']
+            user_id = request.data['user_id']
+            new_state = request.data['new_state']
+        except KeyError as e:
+            return Response({'error': f'Missing required field: {e}'}, status=status.HTTP_400_BAD_REQUEST)
+
+        # Find the reservation based on restaurant_id and user_id
+        try:
+            reservation = Reservation.objects.get(restaurant_id=restaurant_id, user_id=user_id)
+        except Reservation.DoesNotExist:
+            return Response({'error': 'Reservation not found'}, status=status.HTTP_404_NOT_FOUND)
+
+        # Update the reservation state
+        reservation.reservation_state = new_state
+        reservation.save()
+
+        return Response({'success': 'Reservation state updated successfully'}, status=status.HTTP_200_OK)
```

Comment:
The success part of a response should be a boolean. It should be attached to the error responses as false, and true for the successful ones.

Reply:
We can remove the success/error part as a whole and only set the status.
ERROR
```return Response({status=status.HTTP_404_NOT_FOUND)```
SUCCESS
```return Response({status=status.HTTP_200_OK)```

Then check in the frontend using
```if (response.ok) ```

Reply:
truuuue